src/server/plugins/check_version.js

```diff
@@ -0,0 +1,34 @@
+import pluginInit from './plugin_init';
+import { cleanVersion, versionSatisfies } from '../../utils/version';
+import _ from 'lodash';
+
+module.exports = async function (kbnServer, server, config) {
+  const warningMessages = new Set();
+  const plugins = kbnServer.plugins;
+
+  for (let plugin of plugins) {
+    // Plugins must specify their version, and by default that version should match
+    // the version of kibana down to the patch level. If these two versions need
+    // to diverge, they can specify a kibana.version to indicate the version of
+    // kibana the plugin is intended to work with.
+    const version = _.get(plugin, 'pkg.kibana.version', _.get(plugin, 'pkg.version'));
+    const name = _.get(plugin, 'pkg.name');
+
+    if (version === 'kibana') continue;
```

Comment:
Since both the plugin install process and this check use the version.js file, I don't want to move the check into versionSatisfies. I'll clean it up in check_version.js